src/ucp/proto/proto_select.c

```diff
@@ -556,6 +556,57 @@ ucp_proto_select_elem_init_thresh(ucp_worker_h worker,
     return status;
 }
 
+/**
+ * Get map of lanes used in the selected protocols.
+ */
+static ucp_lane_map_t
+ucp_proto_select_get_lane_map(ucp_worker_h worker,
+                              const ucp_proto_select_elem_t *select_elem)
+{
+    ucp_lane_map_t lane_map = 0;
+    size_t range_start, range_end;
+    ucp_proto_query_attr_t query_attr;
+
+    range_end = -1;
+    do {
+        range_start = range_end + 1;
+        if (!ucp_proto_select_elem_query(worker, select_elem, range_start,
+                                         &query_attr)) {
+            break;
+        }
+
+        range_end = query_attr.max_msg_length;
+        lane_map |= query_attr.lane_map;
+    } while (range_end != SIZE_MAX);
+
+    return lane_map;
+}
+
+/**
+ * Activate UCP worker interfaces corresponding to the resources of lanes used
+ * in the selected protocols.
+ */
+static void
+ucp_proto_select_wiface_activate(ucp_worker_h worker,
+                                 const ucp_proto_select_elem_t *select_elem,
+                                 ucp_worker_cfg_index_t ep_cfg_index)
+{
+    ucp_lane_map_t lane_map;
+    ucp_ep_config_key_t *ep_config_key;
+    ucp_lane_index_t lane;
+    ucp_rsc_index_t rsc_index;
+    ucp_worker_iface_t *wiface;
+
+    lane_map      = ucp_proto_select_get_lane_map(worker, select_elem);
+    ep_config_key = &ucs_array_elem(&worker->ep_config, ep_cfg_index).key;
+    ucs_for_each_bit(lane, lane_map) {
+        ucs_assert(lane < UCP_MAX_LANES);
```

Comment:
minor: assertv and print lane and map

Reply:
fixed by https://github.com/openucx/ucx/pull/8986/commits/1b6cb7d0e2a93db5d20aadecb12b6b3b99af67e8